### PR TITLE
Nn 2171 add appointment details

### PR DESCRIPTION
**Pull request by JamieDraperUK**
- Capture initial appointment details and stuff them into session
- Create an initial proposed controller structure
- Create time picker component 
- Convert date picker component into a macro
- Move date picker styles closer to component 
- Add an additional node-sass config to look for styles in the components directory
- Add a users me request and stuff results into session

### CODE REVIEW

File: backend/common-messages.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const serviceUnavailableMessage = 'Sorry, the service is unavailable'
```
Comment (jsrobertson) [inline backend/common-messages.js]: Good idea 👍

File: sass/application.scss [hunk 2]
```diff
@@ -15,22 +17,6 @@ $govuk-page-width: 1170px;
...
-
-.ui-datepicker {
-    margin-top: 20px;
-}
-
-.ui-datepicker table {
-    background: govuk-colour("white") !important;
-}
-
 .ui-widget-content {
```
Comment (jsrobertson) [inline sass/application.scss]: Think the rest of the `ui-` prefixed classes can go in datePicker.scss

File: views/components/timePicker/timePicker.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
+.time-picker-container {
+  display: flex;
+  flex-direction: row;
+
+  .time-picker-hours, time-picker-minutes {
```
Comment (jsrobertson) [inline views/components/timePicker/timePicker.scss]:
```suggestion
  .time-picker-hours, .time-picker-minutes {
```

File: backend/controllers/addAppointmentDetails.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        errors.push({ text: 'Enter a date in DD/MM/YYYY format', href: '#date' })
+
+      if (date && moment(date, DAY_MONTH_YEAR).isBefore(now, 'day')) {
+        errors.push({ text: 'Select a date that is not in the past', href: '#date' })
+      } else {
+        if (startTime && startTime.isBefore(now, 'minutes'))
+          errors.push({ text: 'Select a start time that is not in the past', href: '#start-time' })
+
+        if (endTime && endTime.isBefore(startTime, 'minutes'))
+          errors.push({ text: 'Select a end time that is not in the past', href: '#end-time' })
```
Comment (Robc24) [inline backend/controllers/addAppointmentDetails.js]: really minor - select 'an' end time.

File: backend/controllers/addAppointmentDetails.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+          appointmentTypeDescription: appointmentTypeDetails.text,
+          location,
+          locationDescription: locationDetails.text,
+          date,
+          startTime: startTime.format(DATE_TIME_FORMAT_SPEC),
+          endTime: endTime && endTime.format(DATE_TIME_FORMAT_SPEC),
+          sameTimeAppointments,
+        }
+
+        res.redirect('/upload-file')
```
Comment (Robc24) [inline backend/controllers/addAppointmentDetails.js]: We need to come to consensus on the url.  John has prefixed with `bulk-appointments/` which makes sense but isn't mentioned in the ACs.  I think we should go with bulk-appointments/upload-file (etc).

File: backend/tests/addAppointmentDetails.test.js [hunk 6]
```diff
@@ -0,0 +1,309 @@
+const moment = require('moment')
+const { addAppointmentDetailsFactory } = require('../controllers/addAppointmentDetails')
+const { serviceUnavailableMessage } = require('../common-messages')
+
+const { DATE_TIME_FORMAT_SPEC, DAY_MONTH_YEAR } = require('../../src/dateHelpers')
+
+describe('Add appointment details controller', () => {
+  const bulkdAppointmentService = {}
```
Comment (Robc24) [inline backend/tests/addAppointmentDetails.test.js]: should this have the **d** in it?

File: backend/tests/addAppointmentDetails.test.js [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+        locationDescription: 'location 1',
+        appointmentType: 'app1',
+        appointmentTypeDescription: 'appointment 1',
+        date,
+        startTime: startTime.format(DATE_TIME_FORMAT_SPEC),
+        endTime: endTime.format(DATE_TIME_FORMAT_SPEC),
+        sameTimeAppointments: 'yes',
+      })
+
+      expect(res.redirect).toHaveBeenCalledWith('/upload-file')
```
Comment (Robc24) [inline backend/tests/addAppointmentDetails.test.js]: same as above if we decide to update to `bulk-appointments/`

**Pull request merged**